vpd-tool/src/vpd_tool.cpp

```diff
@@ -0,0 +1,59 @@
+#include "vpd_tool.hpp"
+
+#include "constants.hpp"
+#include "types.hpp"
+#include "utils.hpp"
+
+#include <iostream>
+
+namespace vpd
+{
+void VpdTool::readKeyword(const std::string& i_fruPath,
+                          const std::string& i_recordName,
+                          const std::string& i_keywordName, bool i_onHardware,
+                          const std::string& i_fileToSave)
+{
+    types::DbusVariantType l_keywordValue;
+    if (i_onHardware)
+    {
+        std::cout << "Read from h/w" << std::endl;
+        // TODO: uncomment when implementation of the API goes in
+        /*l_keywordValue = utils::readKeywordFromHardware(
+            constants::vpdManagerService,
+           constants::vpdManagerObjectPath,
+            constants::vpdManagerIntfName, i_fruPath,
+            make_tuple(i_recordName, i_keywordName));*/
+    }
+    else
+    {
+        std::string l_fullObjectPath(constants::baseInventoryPath + i_fruPath);
+
+        l_keywordValue = utils::readDbusProperty(
```

Comment:
readDbusProperty returns empty buffer in case of any exception. can you check if l_keywordValue is empty or not and take action accordingly?

Reply:
if dbus read fails with an exception - empty l_keywordValue is returned. if l_keywordValue is empty, error message at line 55 gets printed which is misleading.